### PR TITLE
OutBuffer: add debug stomping code

### PR DESCRIPTION
**Pull request by WalterBright**
Expose dangling pointer problems by stomping on what they may be left pointing to. This is enabled for the moment, if (when) it is clean I'll enable it only for debug builds.

### CODE REVIEW

**Comment by WalterBright** [comment]
Sweet! DMD fails everywhere with this on. I should have added it long ago.

**Review by marler8997 — APPROVED** [review]
love it

**Comment by rainers** [comment]
Data coming from/going back to the C runtime via malloc/free has usually tools to cover this (e.g. the VC debug runtime stomps memory). Stomping should be useful in setsize/reset, though, where it could also avoid stale GC pointers.

**Comment by WalterBright** [comment]
> Data coming from/going back to the C runtime via malloc/free has usually tools to cover this (e.g. the VC debug runtime stomps memory).

Yes, the only problem is it's different on every platform. Just building it in is easy and portable.

> Stomping should be useful in setsize/reset, though, where it could also avoid stale GC pointers.

Good idea, although I eliminated reset() in another unpulled PR.

File: src/dmd/root/outbuffer.d [hunk 1]
```diff
@@ -18,6 +18,8 @@ import core.stdc.string;
 import dmd.root.rmem;
 import dmd.root.rootobject;
 
+debug = stomp; // flush out dangling pointer problems by stomping on unused memory
```
Comment (Geod24) [inline src/dmd/root/outbuffer.d]: I guess you want to remove this before we merge ?

Comment (WalterBright) [reply src/dmd/root/outbuffer.d]: yes, now that it's working

**Pull request merged**